tests/h/streamer/websocket_test.py

```diff
@@ -94,23 +93,88 @@ def test_socket_send_json_skips_when_terminated(fake_environ, fake_json, fake_so
     assert not fake_socket_send.called
 
 
-def test_handle_message_sets_socket_client_id_for_client_id_messages():
+@pytest.mark.usefixtures('handlers')
+class TestHandleMessage(object):
+
+    def test_closes_connection_for_invalid_json(self):
+        """If we receive invalid JSON, we close the connection."""
+        socket = mock.Mock(spec_set=['close'])
+        message = websocket.Message(socket, payload='gibberish')
+
+        websocket.handle_message(message)
+
+        socket.close.assert_called_once_with(reason='invalid message format')
+
+    def test_uses_unknown_handler_for_missing_type(self, unknown_handler):
+        """If the type is missing, call the `None` handler."""
+        socket = mock.Mock(spec_set=['close'])
+        message = websocket.Message(socket, payload='{"foo":"bar"}')
+
+        websocket.handle_message(message)
+
+        unknown_handler.assert_called_once_with(socket,
+                                                {"foo": "bar"},
+                                                session=None)
+
+    def test_uses_unknown_handler_for_unknown_type(self, unknown_handler):
```

Comment:
Might be good to add a test for when there is no `type` key in the payload? It currently calls the "unknown handler" as well, might be beneficial to have a test for that so we don't accidentally change this behaviour in the future.